personal_trainer/src/source.js

```diff
@@ -112,46 +114,65 @@ function meal_input() {
 //declaration
 let btn_show = document.getElementById("btn_show");
 let date_input = document.getElementById("date_input");
-food_input = JSON.parse(localStorage.getItem("meal"));
+
+//byZiga: here I removed "food_input = JSON..."   and put it inside function (and also renamed variable to food_consumed to not mess with values in food_input function)
+
+//byZiga: set default date for meal output to today   (as seen on Stackoverflow)
+date_input.valueAsDate = new Date();
+
 
 //button and function that lists all meals from local storage
 btn_show.addEventListener("click", function() {meals_output()});
+date_input.addEventListener("change", function() {meals_output()}); //byZiga: show meal output allready when date is picked (user doesnt have to click "show" button) 
 
 //2019-12-08 changed to standalone function for easier calling outside the button
-function meals_output(meal_input_date) {
+function meals_output() {       //byZiga: removed date in brackets because function added meals from different days together
+    let food_consumed = JSON.parse(localStorage.getItem("meal")); //byZiga: var renamed + must be moved inside function, otherwise "TypeError:food_input is null" when user tries to input meal when cache is cleared
+        //by Ziga: previousy this variable was named food_input, but had to be renamed, otherwise problems when cache cleared (it caused input meals to stop working)
     let food_results = document.getElementById("food_results");
     let kcal_counter = 0;
     food_results.innerHTML = "";
-    food_input.forEach(function(item) {
     
-//list meals with selected date
-    if (item.date_time === date_input.value || item.date_time === meal_input_date) { // 2019-12-08 MM added additional processing if I pass date from meal_input() that pushes .value already 
```

Comment:
Tu sem zbrisal drugi pogoj, da ni prikazalo skupnih obrokov tako dneva, ko user vnese input, kot dneva, ki je izbran za prikaz daily vnosov. Spremenjena koda je malce nižje spodaj.
Vidim da je tu veliko označenega z rdečo (ala da naj bi vse to zbrisal), toda spremenil sem le par vrstic (vse so označene s komentarjem //byZiga) - samo GitHub ne prepozna, da so ostale iste, ker je pred njimi spremenjena koda.